### ISSUE TITLE
Support for Google Play APIs

### ISSUE THREAD

**Issue by Fervensimber** [Team-Innis/UtH-Engine#8]
Integration of Google Play Services https://developer.android.com/google/play-services/index.html into engine. Most notably game related apis https://developers.google.com/games/services/cpp/GettingStartedNativeClient and In App Billing https://developer.android.com/google/play/billing/index.html

**Comment by zeitt** [comment]
Achievements: Easy Integration per application. If most teams wants this we will add small wrapper for it.
Saved Games: Will be starting to research this after issue #6 has been fixed.
Events and Quests: Still havent understood what these even are :smiley: 
Leaderboards: Same as with achievements.
Multiplayer: Seems like Google added Realtime MP also. Has to be researched more.

In app billing: Has been split to separate issue #23 & will be starting to look at it after #21 (Ads) has been completed.

**Comment by zeitt** [comment]
Status update:
Achievements & Leaderboards have been implemented.
Google Analytics: W.I.P so it will be implemented. See issue #22 
Other stuff won't be. (No time to get them work properly)
Inapp: Won't be implented. See issue #23  
Ads: Has been implemented. Still needs more testing before pushing to master/develop. See issue #21
